tests/unit/unit-walker.cpp

```diff
@@ -115,8 +174,743 @@ TEST_P(UnitWalkerTests, FunSubstitution)
+    TreeWalker iw(s, false, &passed_map);
+
+    pair<Term, vector<int>> p1;
+    p1 = iw.visit(fullform);
+
+    /* expected_map:
+     * f(y,z) = g(x): <f(y,z) = g(x), []>
+     * f(y,z): <f(y,z) = g(x), [0]>
+     * f: <f(y,z) = g(x), [0,0]>
+     * y: <f(y,z) = g(x), [0,1]>
+     * z: <f(y,z) = g(x), [0,2]>
+     * g(x): <f(y,z) = g(x), [1]>
+     * g: <f(y,z) = g(x), [1,0]>
+     * x: <f(y,z) = g(x), [1,1]>
+     */
+    map<Term, pair<Term, vector<int>>> expected_map;
+    // f(y,z) = g(x) -> <f(y,z) = g(x), []>
+    pair<Term, vector<int>> n0;
+    n0.first = fullform;
+    vector<int> n0path;
+    n0.second = n0path;
+    expected_map[fullform] = n0;
+    // f(y,z) -> <f(y,z) = g(x), [0]>
+    pair<Term, vector<int>> n1;
+    n1.first = fullform;
+    vector<int> n1path = { 0 };
+    n1.second = n1path;
+    expected_map[fyz] = n1;
+    // f -> <f(y,z) = g(x), [0,0]>
+    pair<Term, vector<int>> n2;
+    n2.first = fullform;
+    vector<int> n2path = { 0, 0 };
+    n2.second = n2path;
+    expected_map[f] = n2;
+    // y -> <f(y,z) = g(x), [0,1]>
+    pair<Term, vector<int>> n3;
+    n3.first = fullform;
+    vector<int> n3path = { 0, 1 };
+    n3.second = n3path;
+    expected_map[y] = n3;
+    // z -> <f(y,z) = g(x), [0,2]>
+    pair<Term, vector<int>> n4;
+    n4.first = fullform;
+    vector<int> n4path = { 0, 2 };
+    n4.second = n4path;
+    expected_map[z] = n4;
+    // g(x) -> <f(y,z) = g(x), [1]>
+    pair<Term, vector<int>> n5;
+    n5.first = fullform;
+    vector<int> n5path = { 1 };
+    n5.second = n5path;
+    expected_map[gx] = n5;
+    // g -> <f(y,z) = g(x), [1,0]>
+    pair<Term, vector<int>> n6;
+    n6.first = fullform;
+    vector<int> n6path = { 1, 0 };
+    n6.second = n6path;
+    expected_map[g] = n6;
+    // x -> <f(y,z) = g(x), [1,1]>
+    pair<Term, vector<int>> n7;
+    n7.first = fullform;
+    vector<int> n7path = { 1, 1 };
+    n7.second = n7path;
+    expected_map[x] = n7;
+
+    // check the cache populated by TreeWalker is equivalent to the expected map
+    mapEqual(expected_map, passed_map);
+  }
+}
+
+TEST_P(UnitWalkerTests, FreshVars)
+{
+  // test traversal using IndicatorTreeWalker of formula with repeated values
+
+  SolverConfiguration sc = GetParam();
+  if (sc.is_logging_solver)
+  {
+    // construct full formula, (x+x) = {(y+y) + [(x+2)-(x+x)]}
+    Term x = s->make_symbol("x", bvsort);
+    Term y = s->make_symbol("y", bvsort);
+    Term xpx = s->make_term(BVAdd, x, x);
+    Term ypy = s->make_term(BVAdd, y, y);
+    Term xp2 = s->make_term(BVAdd, x, s->make_term(2, bvsort));
+    Term lmt = s->make_term(BVSub, xp2, xpx);
+    Term rhs = s->make_term(BVAdd, ypy, lmt);
+    Term fullform = s->make_term(Equal, xpx, rhs);
+
+    UnorderedTermPairMap passed_map;
+    IndicatorTreeWalker itw(s, false, &passed_map);
+
+    pair<Term, vector<int>> p1;
+    p1 = itw.visit(fullform);
+
+    /* expected map:
+     * b0: <(x+x)={(y+y)+[(x+2)-(x+x)]} , []>, where b0 corresponds to
```

Comment:
I don't totally follow the format of this comment. Are the keys all the `b*`? I think the `..., where b* corresponds to` is confusing me. It's also possible `clang-format` messed this up (that's happened to me before).

Reply:
Okay this was explained to me offline, makes sense! Only small thing might be to start a new line for each key, value pair. E.g.:
```
* b0: <(x+x)={(y+y)+[(x+2)-(x+x)]} , []>, where b0 corresponds to
* (x+x)={(y+y)+[(x+2)-(x+x)]} 
* b1: <(x+x)={(y+y)+[(x+2)-(x+x)]} , [1]>, where b1 corresponds to 
* {(y+y)+[(x+2)-(x+x)]} 
```

or something like that.